src/proteus/dswx_hls.py

```diff
@@ -4150,6 +4255,53 @@ class in the browse image. False to display PSW-Agg.
     logger.info(f'    mean azimuth angle: {sun_azimuth_angle}')
     logger.info(f'    mean elevation angle: {sun_elevation_angle}')
 
+    # print input HLS product spatial and cloud coverage
+    logger.info(f'data coverage:')
+    if 'INPUT_HLS_PRODUCT_SPATIAL_COVERAGE' in dswx_metadata_dict.keys():
+        hls_product_spatial_coverage = dswx_metadata_dict[
+            'INPUT_HLS_PRODUCT_SPATIAL_COVERAGE']
+        logger.info(f'    input HLS product spatial coverage [%]:'
+                    f' {hls_product_spatial_coverage}')
+
+    if 'INPUT_HLS_PRODUCT_CLOUD_COVERAGE' in dswx_metadata_dict.keys():
+        hls_product_cloud_coverage = dswx_metadata_dict[
+            'INPUT_HLS_PRODUCT_CLOUD_COVERAGE']
+        logger.info(f'    input HLS product cloud coverage [%]:'
+                    f' {hls_product_cloud_coverage}')
+
+    # compute a preliminary cloud layer, i.e., without snow/ice class
+    preliminary_cloud_layer = _compute_preliminary_cloud_layer(
+        fmask, mask_adjacent_to_cloud_mode)
+
+    # create ocean mask
+    if shoreline_shapefile is not None:
+        ocean_mask = _create_ocean_mask(shoreline_shapefile,
+                                        ocean_masking_shoreline_distance_km,
+                                        scratch_dir, geotransform, projection,
+                                        length, width,
+                                        temp_files_list=temp_files_list)
+
+        # update valid_array
+        valid_array = np.logical_and(valid_array, ocean_mask)
+
+    # create variables to compute spatial and cloud coverage
+    n_valid = np.sum(valid_array)
+    n_cloud_and_valid = np.sum(((preliminary_cloud_layer) != 0) & (valid_array))
+    del valid_array
+    total_number_of_pixels = length * width
+
+    # use floor integer to compute spatial and cloud coverage
+    spatial_coverage = int(100 * float(n_valid) / total_number_of_pixels)
+    cloud_coverage = int(100 * float(n_cloud_and_valid) / total_number_of_pixels)
```

Comment:
Is % cloud coverage out of the entire 3660x3660 image? Or % coverage of the valid pixels?

This is perhaps a larger conversation with the group, but I'm starting to have reservations about these new metrics.

For one of the San Francisco Bay granules, here are the numbers:

```
  INPUT_HLS_PRODUCT_CLOUD_COVERAGE=25
  INPUT_HLS_PRODUCT_SPATIAL_COVERAGE=99
  CLOUD_COVERAGE=17
  SPATIAL_COVERAGE=52
```
I'm still trying to put my finger on why this is unsettling, but putting on my user hat, a couple of things stand out right off the bat:
1) Where did the clouds go?
2) Why did we lose half of the granule? To me, this has been a fantastic granule to use as a test case. But when I run queries to STAC-CMR, as a starting point I filter out anything with spatial coverage under 70%, which means I never would have found this granule.

Here's the browse image, for reference:
![OPERA_DSWx-HLS_L8_30_T10SEG_20220119T184616Z_20230130T203109Z_v1 0 0_BROWSE](https://user-images.githubusercontent.com/11642807/215622228-8dfd12bd-6af2-457b-b08c-8cd53083c876.png)

What are your thoughts on this? Is it still the right thing to do?

Reply:
Good point, @nemo794 . You are right, the HLS product defines the `CLOUD_COVERAGE` over valid pixels. I'm updating the code to do the same.

Reply:
Thank you for updating `CLOUD COVERAGE`.

Regarding `SPATIAL_COVERAGE`, I'll just note here that as discussed in yesterday's meeting, the decision is to keep `SPATIAL_COVERAGE` computed as-is, and then add a new metadata field that computes spatial coverage over pixels that are not marked as "ocean" by the ocean mask.